internal/provider/snowflake.go

```diff
@@ -410,6 +404,32 @@ func (p *SnowflakeProvider) ensureTableSchema(schema dm.Model) error {
 	return nil
 }
 
+func (p *SnowflakeProvider) handleSnowflakeError(err error, retryFunc func() error) error {
+	if snowErr, ok := err.(*gosnowflake.SnowflakeError); ok {
+		if snowErr.Number == 390114 {
+			// Authentication token has expired. Re-establish DB connection and retry
+			if err := p.reEstablishConnection(); err != nil {
+				return err
+			}
+			// Retry the operation
+			return retryFunc()
+		}
+	}
+	return err
+}
+
+func (p *SnowflakeProvider) reEstablishConnection() error {
+	p.logger.Error("Snowflake authentication token has expired. Re-establishing connection")
+	p.db.Close()
+	db, err := sql.Open("snowflake", p.url)
+	if err != nil {
+		p.logger.Error("unable to re-create snowflake connection: %w", err)
```

Comment:
i don't think our logger supports %w does it?

Reply:
Oops probably not- tabbed auto-complete without checking. I’ll change when I’m on my computer